### PR TITLE
categorize participants by date of birth instead of self-reported usertype

### PR DESCRIPTION
**Pull request by xustanley**
# Pull Request

## Description
- Replaced self-reported user type selection during registration with automatic assignment based on date of birth.
- Participants < 36 are younger adults (student), 55+ as older adults. 
- Participants aged 36–54 are temporarily marked as null and receive a modal warning upon submission informing them they fall outside the typical age ranges and should contact the admin info@forallages.org after signing up.
- Removed the "Are you registering as a student or older adult?" question from the locked questions list. Existing forms that still have it will show it as deletable in FormBuilder so admins can remove it if they want.

     - The bug that sometimes converts adults to students may be because option choices (e.g. Senior) didn't match the hardcode of Adult and autoconverts it to student, and vice versa. This shouldn't be a problem if DOB is used over the manual selection now.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Related Issues (put task name here from Notion)
[Triage: Older adults showing up as students
](https://app.notion.com/p/h4i/Triage-Older-adults-showing-up-as-students-380197abf07b80818267f83e51672b18?source=copy_link
)

### Code Review Preparation
- [x] I have performed a self-review and reviewed the changed files
- [x] My code follows the project's style guidelines and passes linting
- [x] I have added comments in hard-to-understand areas and updated documentation

### Testing
- [x] I have added relevant tests and they pass locally
- [x] I have tested the changes manually, including edge cases

### Code Quality & Security
- [x] I have formatted my code and removed debugging/unused code
- [x] No TypeScript errors or linting issues
- [x] No sensitive information committed (API keys, passwords, etc.)
- [x] User inputs are validated and sanitized

## Screenshots/Screen recording (If applicable)
https://github.com/user-attachments/assets/ebf7017b-cc03-4fac-bb1d-07d1d69ed126

## Summary by CodeRabbit

* **New Features**
  * Added an age verification modal during registration for applicants outside the typical age range, with “Go Back” and “Continue Signing Up”.
  * Registration now derives participant type from date of birth by calculating age and adjusting the signup flow accordingly.
* **Bug Fixes**
  * Updated FormBuilder locked-question handling so locked status is recalculated to match the current set of supported locked fields, enabling removal of outdated locked items.
  * Switched the default locked question from “user type” to “date of birth”.

### CODE REVIEW

File: src/pages/FormBuilder/useFormEditor.ts [hunk 1]
```diff
@@ -68,6 +67,8 @@ const createId = (prefix: string) => {
 };
 
 const LOCKED_QUESTION_ORDER = LOCKED_QUESTIONS.map((question) => question.lockedKey);
+// Used to track questions that were once locked but then removed from LOCKED_QUESTIONS
+const LOCKED_QUESTION_KEYS = new Set(LOCKED_QUESTIONS.map((q) => q.lockedKey));
```
Comment (evtran4) [inline src/pages/FormBuilder/useFormEditor.ts]: Is this necessary? I don't think locked questions will ever be removed. The app won't be able to function without them. There may be old participant forms without the locked questions, but now and in prod, locked questions should always be answered. 

Let me know if i'm misunderstanding.

File: src/pages/FormBuilder/useFormEditor.ts [hunk 2]
```diff
@@ -135,6 +136,9 @@ const formToState = (form?: Form | null): FormEditorState => {
         locked: section.locked,
         questions: (section.questions ?? []).map((q) => ({
           ...q,
+          // If a question's lockedKey is removed from LOCKED_QUESTIONS, 
+          // treat it as unlocked so admins can delete it from the FormBuilder.
+          locked: q.locked && q.lockedKey ? LOCKED_QUESTION_KEYS.has(q.lockedKey) : q.locked,
```
Comment (evtran4) [inline src/pages/FormBuilder/useFormEditor.ts]: Same comment as above. Is this necessary for your task?

File: src/pages/Registration/RegistrationNew.tsx [hunk 3]
```diff
@@ -646,88 +672,110 @@ const RegistrationNew = ({
...
+          });
+        }
+
 
-          if (current >= max) {
-            return true;
+        if (shouldWaitlist) {
+          const waitlistRef = doc(db, "waitlist", user.uid);
+          await setDoc(waitlistRef, { uid: user.uid, createdAt: serverTimestamp() });
+          navigate("/waiting", { replace: true });
```
[the remark this answers is not in the document]
Comment (evtran4) [reply src/pages/Registration/RegistrationNew.tsx:752]: Is this relevant? I think this comment is speaking to when an admin manually adds a user and fills out the form themselves. Since the "user" is the admin, will it write their UID?

File: src/pages/Registration/RegistrationNew.tsx [hunk 4]
```diff
@@ -875,6 +924,58 @@ const RegistrationNew = ({
...
+                type="button"
+                className={styles.btnBack}
+                onClick={() => {
+                  setShowAgeWarningModal(false);
+                  pendingContinue.current = null;
+                }}
+              >
+                Go Back
+              </button>
+              <button
```
Comment (evtran4) [inline src/pages/Registration/RegistrationNew.tsx:956]: Any chance we can make this the <Button> component in Button.tsx?

File: src/pages/Registration/RegistrationNew.tsx [hunk 5]
```diff
@@ -875,6 +924,58 @@ const RegistrationNew = ({
...
+            <p className={styles.ageModalText}>
+              You are still welcome to sign up! To ensure accurate matching,
+              please reach out to our admin team at{" "}
+              <a href="mailto:info@forallages.org" className={styles.ageModalLink}>
+                info@forallages.org
+              </a>{" "}
+              after registering so they can place you appropriately.
+            </p>
+            <div className={styles.ageModalActions}>
+              <button
```
Comment (evtran4) [inline src/pages/Registration/RegistrationNew.tsx:946]: Any chance we can make this the <Button> component in Button.tsx?

**Review by evtran4 — COMMENTED** [review]
Looks pretty good overall! Some minor comments

File: src/pages/Registration/RegistrationNew.tsx (hunk 4 above)
Comment (xustanley) [reply src/pages/Registration/RegistrationNew.tsx:956]: Hmmm probably worth making all buttons follow that eventually. It doesn't have a type prop, which we'd need for the non-submit buttons to prevent them from accidentally triggering form submission. Only AdminCreator uses it rn, and that's for action buttons rather than form submission- everything else still uses plain <button> elements. @coderabbitai

File: src/pages/Registration/RegistrationNew.tsx (hunk 5 above)
Comment (xustanley) [reply src/pages/Registration/RegistrationNew.tsx:946]: comment above

File: src/pages/Registration/RegistrationNew.tsx (hunk 4 above)
Comment (evtran4) [reply src/pages/Registration/RegistrationNew.tsx:956]: Good catch! Thanks

**Review by evtran4 — APPROVED** [review]
